### PR TITLE
Create configuration files for Client and Server

### PR DESCRIPTION
**Pull request by plutonik-a**
This PR will provide two separate client-side and server-side configuration files in JSON format.

The client-side configuration is static, bundled with the client code and contains:

- Strings for error page and toasts containing error/success/info messages
- Frontend server URL
- GitHub API endpoints (e.g. for easily creating URLs in test suite)
- Default text modules for Petal Commit messages

The server-side configuration is loaded dynamically and contains:

- API URL
- enabling/disabling CORS
- Petal App User / Email

### CODE REVIEW

File: packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts [hunk 1]
```diff
@@ -18,13 +18,21 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
...
 import { retry } from "@octokit/plugin-retry";
-dotenv.config();  // Load environment variables from .env file 
+import * as fs from 'fs';
+
+dotenv.config();  // Load environment variables from .env file
+
+/**
+ * Load the configuration
+ */
+const config = JSON.parse(fs.readFileSync('./server-config.json', 'utf8'));
```
Comment (marmoure) [inline packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts:30]: @plutonik-a Typescript allows us to import json directly without having to use the fs module
So please change this to 
```ts
import config from './server-config.json`
```
And add this in the top level tsconfig
```json
    "resolveJsonModule": true,                /* Allows importing JSON files */
```

File: packages/prosemirror-lwdita-backend/src/server.ts [hunk 2]
```diff
@@ -17,10 +17,21 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
...
+import cors from 'cors';
+import * as fs from 'fs';
 
 const app = express();
 app.use(express.json());
 
+/**
+ * Load the configuration
+ */
+const config = JSON.parse(fs.readFileSync('./server-config.json', 'utf8'));
```
Comment (marmoure) [inline packages/prosemirror-lwdita-backend/src/server.ts:29]: Same as the other json config comment

File: packages/prosemirror-lwdita-backend/tsconfig.json [hunk 3]
```diff
@@ -5,7 +5,9 @@
     "rootDir": "src",
     "outDir": "dist",
     "module": "Node16",
-    "moduleResolution": "Node16"
+    "moduleResolution": "Node16",
+    "resolveJsonModule": true,
+    "esModuleInterop": true
```
Comment (marmoure) [inline packages/prosemirror-lwdita-backend/tsconfig.json:10]: Why are we setting `esModuleInterop` to true??
`resolveJsonModule` Should be pushed to the top level config.

File: packages/prosemirror-lwdita/src/commands.ts [hunk 4]
```diff
@@ -244,16 +245,16 @@ export function renderPrDialog(ghrepo: string, source: string, branch: string, u
...
-        console.log('The document has been published to GitHub.');
-        console.log('pr url:', prURL);
-        // Show a user notification with the successful result and a link to the PR
-        showPublicationResultSuccess(prURL)
-      }).catch((error: Error) => {
-        console.error('Error:', error);
-        // show error dialog
-        showPublicationResultError(error.message)
-      });
+        .then((prURL: string) => {
```
Comment (marmoure) [inline packages/prosemirror-lwdita/src/commands.ts:248]: Why is there changes in here? I don't see anything changed in the code, is it just whitespace?

File: packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts (hunk 1 above)
Comment (plutonik-a) [reply packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts:30]: Due to import issues in this specific package it seemed to be the only viable option.

File: packages/prosemirror-lwdita-backend/src/server.ts (hunk 2 above)
Comment (plutonik-a) [reply packages/prosemirror-lwdita-backend/src/server.ts:29]: @marmoure See above

File: packages/prosemirror-lwdita-backend/tsconfig.json (hunk 3 above)
Comment (plutonik-a) [reply packages/prosemirror-lwdita-backend/tsconfig.json]: @marmoure I have removed it from this config, but FYI, we're using this option already by importing it from the default config `@tsconfig/recommended/tsconfig.json`.
Fixed with 0cf999e

File: tsconfig.json [hunk 5]
```diff
@@ -12,5 +12,6 @@
     "moduleResolution": "node10",             /* NOTE(AR) just making it explicit that this is the tsc compilation strategy for module resolution that we are currently using */
     "rootDir": "src",
     "outDir": "dist",                         /* Redirect output structure to the `dist/` directory. */
+    "resolveJsonModule": true
```
Comment (marmoure) [inline tsconfig.json:15]: Since we are no longer importing `json`, this flag is not needed.

Comment (plutonik-a) [reply tsconfig.json:15]: @marmoure Sure it does, we are importing JSON.

File: packages/prosemirror-lwdita/src/commands.ts (hunk 4 above)
Comment (plutonik-a) [reply packages/prosemirror-lwdita/src/commands.ts:248]: @marmoure Yes, seems to be a whitespace change.

File: tsconfig.json (hunk 5 above)
Comment (marmoure) [reply tsconfig.json:15]: @plutonik-a I just doubled checked, I don't see any json import in your changes, please point it to me.

**Review by marmoure — APPROVED** [review]
LGTM

**Pull request merged**